### PR TITLE
add refresh id token logic to zli

### PR DESCRIPTION
**Pull request by lgmugnier**
## Description of the change

The daemon did not refresh the id token, so when there was no zli activity, the id token would expire without any way to refresh it.  Now, the daemon refreshes the token whenever it sends a new Syn, which is when a datachannel is opened and then whenever it receives a keysplitting error from the agent (up to a certain number of times).

## Type of change

- [x] Bug fix (non-breaking change that fixes an issue)
- [ ] Refactor (code change with no functionality change)
- [ ] Enhancement (minor change below the level of a feature)
- [ ] New feature (non-breaking change that adds functionality)
- [ ] Remove feature (removes a feature we don't support anymore)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Related JIRA tickets

Relates to JIRA: CWC-912

## Relevant release note information

Release Notes:

## Potential Security Impacts

Does this PR have any security impact?
Does it fix a security issue?
Does it add attack surface?
Why do we think this change is safe?

### Development

- [ ] Does the code changed/added as part of this pull request have test coverage?
- [ ] Do all tests related to the changed code pass in development?
- [ ] Have you ensured that at least one other person has tested your code?
- [ ] Have you tested the code?
- [ ] Have you attached any pictures (if relevant)?

### CODE REVIEW

File: kube-go-assets/bctl/daemon/keysplitting/keysplitting.go [hunk 1]
```diff
@@ -179,6 +182,11 @@ func (k *Keysplitting) BuildSyn(action string, payload []byte) (ksmsg.Keysplitti
 }
 
 func (k *Keysplitting) buildBZCert() (bzcrt.BZCert, error) {
+	// update the id token by calling the passed in zli command
+	if err := exec.Command("bash", "-c", k.refreshTokenCommand).Run(); err != nil {
```
Comment (sidpremkumar) [inline kube-go-assets/bctl/daemon/keysplitting/keysplitting.go:186]: So this assumes that the shell is bash, can we remove "bash" and "-c" and will the command still work? What if the user only has `sh` installed or something

Comment (lgmugnier) [reply kube-go-assets/bctl/daemon/keysplitting/keysplitting.go:186]: testing...

File: src/handlers/tunnel/start-kube-daemon.handler.ts [hunk 2]
```diff
@@ -160,6 +164,21 @@ async function copyExecutableToLocalDir(logger: Logger, configPath: string): Pro
     // Helper function to copy the Daemon executable to a local dir on the file system
     // Ref: https://github.com/vercel/pkg/issues/342
 
+    const WINDOWS_DAEMON_PATH : string = 'kube-go-assets/bctl/daemon/daemon-windows';
+    const LINUX_DAEMON_PATH   : string = 'kube-go-assets/bctl/daemon/daemon-linux';
+    const MACOS_DAEMON_PATH   : string = 'kube-go-assets/bctl/daemon/daemon-macos';
+
+    let prefix = '';
```
Comment (sidpremkumar) [inline src/handlers/tunnel/start-kube-daemon.handler.ts:171]: So this is a common patter we follow, can we make a helper function called ~ `isPkgProcess()` that will return the result of `pkgProcess`

```
function isPkgProcess(): bool {
const process1 = <any>process;
return process1.pkg;
}
```

Comment (lipmas) [reply src/handlers/tunnel/start-kube-daemon.handler.ts:171]: yes sorry i was planning on doing that then forgot :facepalm: :

File: kube-go-assets/bctl/daemon/keysplitting/keysplitting.go (hunk 1 above)
Comment (lgmugnier) [reply kube-go-assets/bctl/daemon/keysplitting/keysplitting.go:186]: experimenting...

Comment (lgmugnier) [reply kube-go-assets/bctl/daemon/keysplitting/keysplitting.go]: 2384f65

**Review by lipmas — APPROVED** [review]

**Pull request merged**